menu/menu_setting.c

```diff
@@ -17860,6 +17925,164 @@ static bool setting_append_list(
+         (*list)[list_info->index - 1].action_left   = &setting_bool_action_left_with_refresh;
+         (*list)[list_info->index - 1].action_right  = &setting_bool_action_right_with_refresh;
+         CONFIG_STRING(
+               list, list_info,
+               settings->arrays.cloud_storage_google_client_id,
+               sizeof(settings->arrays.cloud_storage_google_client_id),
+               MENU_ENUM_LABEL_CLOUD_STORAGE_GOOGLE_CLIENT_ID,
+               MENU_ENUM_LABEL_VALUE_CLOUD_STORAGE_GOOGLE_CLIENT_ID,
+               "",
+               &group_info,
+               &subgroup_info,
+               parent_group,
+               general_write_handler,
+               general_read_handler);
+         CONFIG_STRING(
+               list, list_info,
+               settings->arrays.cloud_storage_google_client_secret,
+               sizeof(settings->arrays.cloud_storage_google_client_secret),
+               MENU_ENUM_LABEL_CLOUD_STORAGE_GOOGLE_CLIENT_SECRET,
+               MENU_ENUM_LABEL_VALUE_CLOUD_STORAGE_GOOGLE_CLIENT_SECRET,
+               "",
+               &group_info,
+               &subgroup_info,
+               parent_group,
+               general_write_handler,
+               general_read_handler);
+
+         CONFIG_BOOL(
+               list, list_info,
+               &settings->bools.cloud_storage_onedrive_default_creds,
+               MENU_ENUM_LABEL_CLOUD_STORAGE_ONEDRIVE_DEFAULT_CREDS,
+               MENU_ENUM_LABEL_VALUE_CLOUD_STORAGE_ONEDRIVE_DEFAULT_CREDS,
+               true,
+               MENU_ENUM_LABEL_VALUE_OFF,
+               MENU_ENUM_LABEL_VALUE_ON,
+               &group_info,
+               &subgroup_info,
+               parent_group,
+               general_write_handler,
+               general_read_handler,
+               SD_FLAG_NONE);
+         (*list)[list_info->index - 1].action_ok     = &setting_bool_action_left_with_refresh;
+         (*list)[list_info->index - 1].action_left   = &setting_bool_action_left_with_refresh;
+         (*list)[list_info->index - 1].action_right  = &setting_bool_action_right_with_refresh;
+         CONFIG_STRING(
+               list, list_info,
+               settings->arrays.cloud_storage_onedrive_client_id,
+               sizeof(settings->arrays.cloud_storage_onedrive_client_id),
+               MENU_ENUM_LABEL_CLOUD_STORAGE_ONEDRIVE_CLIENT_ID,
+               MENU_ENUM_LABEL_VALUE_CLOUD_STORAGE_ONEDRIVE_CLIENT_ID,
+               "",
+               &group_info,
+               &subgroup_info,
+               parent_group,
+               general_write_handler,
+               general_read_handler);
+
+         END_SUB_GROUP(list, list_info, parent_group);
+         END_GROUP(list, list_info, parent_group);
+         break;
+      case SETTINGS_LIST_CLOUD_STORAGE_PROVIDER:
+         {
+            const char *loading_str = "Loading...";
+
+            strcpy(cloud_storage_authorize_status_str, loading_str);
```

Comment:
Why not just a literal?
Or at least re-use it in CONFIG_STRING call

Reply:
Re-used this for now. It will need more work later.